Exercise11_2.R

```diff
@@ -0,0 +1,39 @@
+#Create files to store motifs
+file.create("motif1.fasta")
+file.create("motif2.fasta")
+
+#create vector that has each line of fasta as an elemnt
+Line <- scan("motifsort.fasta", what = character(), sep = "\n")
+
+#create vector of sequences to search, third element is just a dummy so that when it loops the 3rd time
+#it wont find anything but I can keep everything in a single for-loop section.
+seqs <- c("AKKPRVZE", "AAQWWRNYGG", "StuartJones")
+
+
+#look through the loop three times
+for (j in 1:3){
+  #start searching on the second line so that when I delete things it won't try to delete Line[0] (doesn't exist)
+  for (i in 2:length(Line)){
+    #search for string specified by vector
+    if (str_detect(Line[i], seqs[j]) == TRUE){
+      #Write the line that has the string it in and the immediately previous line (the sequence number) to a file
+      write(Line[i-1], file=paste("motif",j,".fasta", sep = ""),append=TRUE)
+      write(Line[i], file=paste("motif",j,".fasta", sep = ""),append=TRUE)
+      #replace those strings with blanks, tried to set to NULL but ran into some error, this is just as good
+      Line[i-1] <- ""
+      Line[i]<-""
+    }
+    #The third time you run the loop every line that has the motif we're looking for should be turned into blanks
+    #leaving only the lines that didnt have either motif
+  if (j == 3){
+    #if a line isn't a blank..
+    if(Line[i] != ""){
+      #write that line into a file
+    write(Line[i], file=paste("motif_other.fasta", sep = ""),append = TRUE)
+    }
+  }
+  }
+}
```

Comment:
good job. another approach could be:

Reply:
library(stringr)

inFile=scan("motifsort.fasta",sep="\n",what=character())

motif1=file("motif1.fasta",open="w")
motif2=file("motif2.fasta",open="w")
other=file("nomotif.fasta",open="w")

for(i in 1:length(inFile)){
  if(str_detect(inFile[i],">")){
    seqID=inFile[i]
  }else{
    if(str_detect(inFile[i],"AKKPRVZE")){
      writeLines(seqID,motif1)
      writeLines(inFile[i],motif1)
    }else if(str_detect(inFile[i],"AAQWWRNYGG")){
      writeLines(seqID,motif2)
      writeLines(inFile[i],motif2)
    }else{
      writeLines(seqID,other)
      writeLines(inFile[i],other)
    }
  }
}


close(motif1)
close(motif2)
close(other)